cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceSourceContext.java

```diff
@@ -56,6 +58,11 @@ public Input call() throws Exception {
     inputNames.add(trackableInput.getAlias());
   }
 
+  @Override
+  public boolean isPreviewEnabled() {
```

Comment:
This is to expose the preview to the plugin itself in the PrepareRun() method so that plugin will be responsible to call this method.